File: CODIGOS6/104.scala
import scala.io.Source

object Main {
  def main(args: Array[String]): Unit = {
    val eggs = Source.fromFile("input.txt").getLines().flatMap(_.split(" ")).map(_.toInt)
    val touching = Array.fill(20)(Array.fill(20)(0))

    for (i <- 0 until 20) {
      val line = Source.fromFile("input.txt").getLines().next().split(" ")
      val eggIndex = line(0).toInt
      val neighbors = line.drop(1).map(_.toInt)
      touching(eggIndex)(neighbors: _*) = 1
      touching(neighbors: _*)(eggIndex) = 1
    }

    val ans = for (i <- 0 until 20) yield {
      if (all(eggs != 1 | touching(i).exists(_ == 1))) i
    }

    println(ans.mkString(" "))
  }
}

File: CODIGOS6/104.r
eggs <- as.numeric(readLines(file("input.txt")))
touching <- matrix(0, 20, 20)

for (i in 1:20) {
  line <- readLines(file("input.txt"))
  egg_index <- as.numeric(line[1])
  neighbors <- as.numeric(line[2:(length(line)-1)])
  touching[egg_index, neighbors] <- 1
  touching[neighbors, egg_index] <- 1
}

ans <- c()
for (i in 1:20) {
  if (all(eggs != 1 | touching[i, ])) {
    ans <- c(ans, i)
  }
}

print(ans)
